services/frontend/adapters/outbound/context_box_connector.adapter.go

```diff
@@ -0,0 +1,101 @@
+package outboundAdapters
+
+import (
+	"errors"
+	"time"
+
+	grpc_retry "github.com/grpc-ecosystem/go-grpc-middleware/retry"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/connectivity"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/berops/claudie/proto/pb"
+	cbox "github.com/berops/claudie/services/context-box/client"
+)
+
+// communicates with the gRPC server of context-box microservice
+type ContextBoxConnector struct {
+	connectionUri string
+
+	// grpcConnection is the underlying gRPC connection to context-box microservice.
+	grpcConnection *grpc.ClientConn
+
+	// GrpcClient is a gRPC client connection to context-box microservice.
+	GrpcClient pb.ContextBoxServiceClient
+}
+
+func NewContextBoxConnector(connectionUri string) *ContextBoxConnector {
```

Comment:
Ditto, please add some comments above functions

Reply:
Not a big deal, but comments above functions usually start with the function name, i.e. how you did it in here.

(This applies to more places, not only here)
```go
// PerformHealthCheck checks health of the underlying gRPC connection to context-box microservice
func (c *ContextBoxConnector) PerformHealthCheck() error {
```